pennylane/_qubit_device.py

```diff
@@ -759,9 +759,15 @@ def expval(self, observable, shot_range=None, bin_size=None):
             )
             return probs[idx]
 
+        # exact expectation value
         if self.shots is None:
-            # exact expectation value
-            eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            try:
+                eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            except NotImplementedError as e:
+                raise ValueError(
+                    f"Cannot compute expectations of {observable.name} observables."
+                ) from e
+
```

Comment:
Previously, the line assumed that all observables have `eigvals` - but Hamiltonians don't. It is safer to try here. 

In future I want to change the logic everywhere to either ask for eigvals or a sparse representation!